studygroups/tasks.py

```diff
@@ -665,3 +667,21 @@ def send_out_community_digest():
 
     if iso_week % 3 == 0:
         send_community_digest(start_date, end_date)
+
+@shared_task
+def refresh_instagram_token():
+    """ Refresh long-lived user access token for Instagram Basic Display API """
+    """ https://developers.facebook.com/docs/instagram-basic-display-api/reference/refresh_access_token """
+    url = "https://graph.instagram.com/refresh_access_token?grant_type=ig_refresh_token&access_token={}".format(settings.INSTAGRAM_TOKEN)
+    response = get_json_response(url)
+    new_token = response.get("access_token", None)
+    request_error = response.get("error", None)
+
+    if new_token:
+        os.environ["INSTAGRAM_TOKEN"] = new_token
```

Comment:
@s-kennedy this will only be visible for the current process and will be lost after the process restarts. The code is executed in one of several workers processes created by gunicorn. The processes are restarted when we update, when there is an unrecoverable error and maybe and maybe under a few other circumstances.

Reply:
ah ok I should have known it wouldn't be so easy :( is there another way to do this then? or maybe make it part of the startup script?

Reply:
I think it would be best to store the instagram token in the database. It seems silly to have a table only for one instagram token, but any other way I can think of would break when we do updates or be more complicated.

Most other services using OAuth allows you to generate an token for API use coupled to your app and account that doesn't expire unless revoked.